### DISCUSSION TITLE
Build tinycurl for FreeRTOS

### DISCUSSION THREAD

**Discussion by bil-ash** [curl/curl#6404] [Ideas]
Please post a blog article explaining how to build tinycurl for FreeRTOS because the documentation (TINY-CURL.md file) mentions only how to build for Linux. Also would be kind of you if you could provide a demo in the FreeRTOS demo style,ex( https://github.com/FreeRTOS/FreeRTOS/tree/master/FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix ). In the blog article please also mention if these FreeRTOS libraries are required for building tinycurl-FreeRTOSPlusTCP,coreHTTP,coreJSON,etc

**Comment by bagder** [comment]
I did the necessary changes for the code to build with the FreeRTOS TCP stack in the tiny-curl branch but I didn't go the extra mile to make it into a full demo in their style. I used my own custom Makefile for the build, and I basically just built all libcurl source files with it together with the rest of the system. It's really not that complicated.

[Tiny-curl](https://curl.se/tiny/) is an effort I've pushed for in an attempt to drum up some interest from people using these RTOSes and others interested in an as small libcurl as possible for their targets  - paying for it by getting somewhat limited libcurl functionality. But my work isn't taken 100% all the way for all targets and I don't intend to do that either on my own. Either people step up and help out to make it happen or pay to have it done. Otherwise I will decide on my own how I spend my spare free time. After all, I earn my living on people paying me (via wolfSSL) for curl support. I cannot do everything in curl out the goodness of my heart as it doesn't bring enough food for my family.

**Comment by bil-ash** [comment]
Your comment helped me know 2 things-
1. **Tinycurl for FreeRTOS is dependent only on the FreeRTOS TCP stack.** Please do put it in the docs as it may turn out to be helpful for someone else like me.
2. **The necessary changes for tinycurl were made in https://github.com/curl/curl/tree/curl-tiny-7.72 branch.** Its the 12th branch in the repository(website),so I couldn't spot it earlier.Also, before starting the discussion, I had downloaded the tarball of tinycurl,I noticed the config-freertos.h header file but couldn't figure out the files in which (c)code changes were made and what were the changes. 

Thanks for both.
